src/main/scala/beam/sim/LoggingParallelEventsManager.scala

```diff
@@ -56,4 +72,53 @@ class LoggingParallelEventsManager @Inject()(config: Config) extends EventsManag
         logger.error(s"Method '$what' failed with: ${t.getMessage}. Stacktrace: $st", t)
     }
   }
+
+  private def createDedicatedHandler: Future[Unit] = {
+    logger.info("Running dedicated blocking handler...")
+    Future {
+      // Let's go for now with blocking approach because it's safe CPU and pretty good for
+      // the scenario when it's many writers but only one reader
+      handleBlocking()
+    }(ec)
+  }
+
+  private def handleNonBlocking(): Unit = {
+    // This is going to waste some CPU
+    while (!isFinished.get()) {
+      val event = nonBlockingQueue.poll()
+      if (null != event)
+        eventManager.processEvent(event)
+    }
+    // We have to consumer the whole queue
+    var isDone = false
+    val start = System.currentTimeMillis()
+    while (!isDone) {
+      val event = nonBlockingQueue.poll()
+      if (null != event)
+        eventManager.processEvent(event)
+      else
+        isDone = true
+    }
+    val end = System.currentTimeMillis()
+    logger.info("Stopped dedicated handler(handleNonBlocking). Took {} ms to process after stop", end - start)
+  }
+
+  private def handleBlocking(): Unit = {
+    while (!isFinished.get()) {
+      val event = blockingQueue.poll(1, TimeUnit.SECONDS)
+      if (null != event)
+        eventManager.processEvent(event)
+    }
+    // We have to consumer the whole queue
+    var isDone = false
+    val start = System.currentTimeMillis()
+    while (!isDone) {
```

Comment:
⚠️ Duplicated code
The lines `95-101` and `115-121` are duplicated. Consider extracting a method.